### DISCUSSION TITLE
Question about A matrix in eigenvalue analysis

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#438] [Q&A] [answered]
[Update] See also https://github.com/CURENT/andes/discussions/153, https://github.com/CURENT/andes/discussions/379

@shebuxin Hello Buxin,

In the control theory, it is common to describe a system in the below formulation (1):

$$
\begin{equation}
\begin{aligned}
\dot{x}(t) &= A x(t) + B u(t) \\
y(t) &= C x(t) + D u(t)
\end{aligned}
\end{equation}
$$

In ``EIG``, the A matrix can be derived by the function ``EIG.calc_As()`` as described in [documentation](https://docs.andes.app/en/latest/_generated/_generated/andes.routines.eig.EIG.calc_As.html) in the below formulation (2):

$$
\begin{equation}
\begin{aligned}
T \dot{x} &= f(x, y) \\
0 &= g(x, y)
\end{aligned}
\end{equation}
$$

And A matrix is calculated in equation (3):

$$
\begin{equation}
\begin{aligned}
A_s &= T^{-1} (f_x - f_y * g_y^{-1} * g_x)
\end{aligned}
\end{equation}
$$

How to understand these two formats? Are they the same thing?

Regards,
Jinning

**Comment by jinningwang** [comment]
To my understanding, a system in the steady state is an autonomous system that can be described as below (1):

$$
\begin{equation}
\begin{aligned}
x &= f(x, y) \\
0 &= g(x, y)
\end{aligned}
\end{equation}
$$

We can compute the Jacobian matrices of the system below (2):

$$
A=\left.\frac{\partial f(x)}{\partial x}\right|_{x=x_0}, \quad B=0, \quad C=\left.\frac{\partial g(x)}{\partial x}\right|_{x=x_0}, \quad D=0
$$

Here, $x_0$ is the operating point of the system. The resulting classical state space model can be written as (3):

$$
\begin{aligned}
\dot{x}(t) & =A x(t) \\
y(t) & =C x(t)
\end{aligned}
$$

Please correct me if I misunderstood anything. @shebuxin 

Regards,
Jinning

**Reply by shebuxin** [reply]
The way to calculate the Jacobian matrix seems not correct. It should be 

$$\begin{equation} \begin{aligned} A = f_x - f_y g_y^{-1} g_x \end{aligned}\end{equation}$$

Then, the small signal state space model is

$$ \begin{aligned} \Delta \dot{x}(t) & =A \Delta x(t) \end{aligned} $$

And the output $\Delta y$ could be customized and defined.

**Comment by shebuxin** [comment, marked as answer]
Hello, Jinning,

Based on my understanding, equation (1) describes an ODE system with $\dot{x}$ and $y$  being explictly expressed by state variables, while for power flow equations (phasor model) in power system, they are implicty related to state variables $x$. Hence, we may have DAE system described in equation (2).

For system (1), Jocobian matrix is $A$; and for system (2), Jicobian matrix is expressed in (3).

If we build up EMT model of power system, the newtwork side phasor power flow will be replaced by DE, then it is possible to represent power system by ODE.
